slycot/analysis.py

```diff
@@ -121,28 +124,21 @@ def ab01nd(n,m,A,B,jobz='N',tol=0,ldwork=None):
     if jobz == 'N':
         out = _wrapper.ab01nd_n(n,m,A,B,tol=tol,ldwork=ldwork)
         if out[-1] < 0:
-            error_text = "The following argument had an illegal value: "+arg_list[-out[-1]-1]
-            raise ValueError(error_text)
+            raise SlycotParameterError(None, out[-1], arg_list)
```

Comment:
Looks a bit ugly and hackish

Would it be an idea to use named parameters?
```
raise SlycotParameterError(info=out[-1], arglist=arg_list)
```
Or call 
```
SlycotParameterError(arg_list, out[-1])
```
then detect that the first parameter is not a string and compose a message?

Reply:
Thanks for looking at this!

I wanted to make it consistent with the `("msg", info)` order of parameters. 

I personally don't like the overloading of a single constructor but that would be possible.

Another option: Have an additional subclass derived from `SlycotParameterError` with `SlycotGenericParameterError(arg_list, info)`. Maybe there is a better idea for the name, though.

Or get rid of the ever same
```python

if info < 0:
    raise SlycotParameterError(....)
```

and replace it with a call of

```python

def check_illegal_parameter(arg_list, info):
    if info < 0:
         message = ("The following argument had an illegal value: {}"
                    "".format(arg_list[-info-1]))
         raise SlycotParameterError(message, info)
```

Reply:
That last solution is the way to go!

Reply:
I am getting another idea, can we format the messages from the "Raises" section in the functions docstring? That would force good docstrings, and eliminate the typing!

```python
def verify_slycot_return(arg_list, docstring, info):
    if info < 0:
         message = ("The following argument had an illegal value: {}"
                    "".format(arg_list[-info-1]))
         raise SlycotParameterError(message, info)
    elif info > 0:
         # process the docstring for the error message
         # ....
```

Reply:
p.s. I am working on this now...